pkg/templates/nstemplatetiers/nstemplatetier_generator_test.go

```diff
@@ -48,20 +48,31 @@ func TestCreateOrUpdateResources(t *testing.T) {
 				}
 				return clt.Client.Create(ctx, obj)
 			}
-			// verify that no "advanced" nor "basic" nor "team" NSTemplateTier resources exist prior to creation
-			for _, tierName := range []string{"advanced", "basic", "team"} {
-				tier := toolchainv1alpha1.NSTemplateTier{}
-				err = clt.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: tierName}, &tier)
-				require.Error(t, err)
-				assert.IsType(t, metav1.StatusReasonNotFound, apierrors.ReasonForError(err))
-			}
+			// verify that no NSTemplateTier resources exist prior to creation
+			nsTmplTiers := toolchainv1alpha1.NSTemplateTierList{}
+			err = clt.List(context.TODO(), &nsTmplTiers, client.InNamespace(namespace))
+			require.NoError(t, err)
+			require.Empty(t, nsTmplTiers.Items)
+			// verify that no TierTemplate resources exist prior to creation
+			tierTmpls := toolchainv1alpha1.TierTemplateList{}
+			err = clt.List(context.TODO(), &tierTmpls, client.InNamespace(namespace))
+			require.NoError(t, err)
+			require.Empty(t, tierTmpls.Items)
+
 			assets := nstemplatetiers.NewAssets(testnstemplatetiers.AssetNames, testnstemplatetiers.Asset)
 
 			// when
 			err := nstemplatetiers.CreateOrUpdateResources(s, clt, namespace, assets)
 
 			// then
 			require.NoError(t, err)
+
+			// verify that 14 TierTemplates were created (see `testnstemplatetiers.AssetNames`)
+			tierTmpls = toolchainv1alpha1.TierTemplateList{}
+			err = clt.List(context.TODO(), &tierTmpls, client.InNamespace(namespace))
+			require.NoError(t, err)
+			require.Len(t, tierTmpls.Items, len(testnstemplatetiers.AssetNames())-1) // exclude `metadata.yml` from the AssetNames, it does not result in a TemplateTier resource
```

Comment:
it would be probably also good to verify the names of the TierTemplates - not only their number

Reply:
done in 50010d3